src/python/ensembl/io/genomio/gff3/restructure.py

```diff
@@ -0,0 +1,253 @@
+
+    Raises an exception if the feature structure is not recognized.
+    """
+    counts = _get_feat_counts(gene)
+    if not counts.get("mRNA") or not counts.get("CDS"):
+        return
+    if counts.get("mRNA") > 1:
+        raise GFFParserError(
+            f"Can't fix gene {gene.id}: contains several mRNAs and CDSs, all children of the gene"
+        )
+
+    # First, count the types
+    mrnas = []
+    cdss = []
+
+    gene_subf_clean = []
+    for subf in gene.sub_features:
+        if subf.type == "mRNA":
+            mrnas.append(subf)
+        elif subf.type == "CDS":
+            cdss.append(subf)
+        else:
+            gene_subf_clean.append(subf)
+
+    mrna = mrnas[0]
+
+    # Check if there are exons (or CDSs) under the mRNA
+    sub_cdss = []
+    sub_exons = []
+    for subf in mrna.sub_features:
+        if subf.type == "CDS":
+            sub_cdss.append(subf)
+        elif subf.type == "exon":
+            sub_exons.append(subf)
+
+    # Check sub CDSs
+    if sub_cdss:
+        raise GFFParserError(f"Gene {gene.id} has CDSs as children in both the gene and mRNA")
+
+    # If there are exons, check they overlap with the CDSs
+    _check_sub_exons(mrna, cdss, sub_exons)
+
+    # No more issues? move the CDSs, and add any new exons
+    mrna.sub_features += cdss
+    # And remove them from the gene
+    gene.sub_features = gene_subf_clean
+    gene.sub_features.append(mrna)
+    logging.debug(f"Gene {gene.id}: moved {len(cdss)} CDSs to the mRNA")
+
+
+def _check_sub_exons(mrna: SeqFeature, cdss: SeqFeature, sub_exons: List[SeqFeature]) -> None:
+    """Check that the exons of the mRNA and the CDSs match.
+    If there are no exons, create them from the CDSs.
+    """
+
+    new_sub_exons = []
+    if sub_exons:
+        # Check that they match the CDS outside
+        if len(sub_exons) == len(cdss):
+            # Now that all coordinates are the same
+            coord_exons = [f"{exon.location}" for exon in sub_exons]
+            coord_cdss = [f"{cds.location}" for cds in cdss]
+
+            if coord_exons != coord_cdss:
+                raise GFFParserError(f"Gene CDSs and exons under the mRNA {mrna.id} do not match")
+        else:
+            raise GFFParserError(
+                f"Gene CDSs and exons under the mRNA {mrna.id} do not match (different count)"
+            )
+    else:
+        # No exons in the mRNA? Create them with the CDS coordinates
+        for cur_cds in cdss:
+            sub_exon = SeqFeature(cur_cds.location, type="exon")
+            sub_exon.sub_features = []
+            new_sub_exons.append(sub_exon)
+    mrna.sub_features += new_sub_exons
+
+
+def remove_extra_exons(gene: SeqFeature) -> None:
+    """Remove duplicated exons existing in both the gene and the mRNAs.
+
+    This is a special case where a gene contains proper mRNAs, etc. but also
+    extra exons for the same features. Those exons usually have an ID starting with
+    "id-", so that's what we use to detect them.
```

Comment:
```suggestion
    This is a special case where a gene contains proper mRNAs, etc. but also extra exons for the same
    features. Those exons usually have an ID starting with "id-", so that is what we use to detect them.
    
    Args:
        gene: Gene feature to update.

    Raises:
        GFFParserError: If not all exons of this gene start with "id-".
```